strawberry/schema/schema_converter.py

```diff
@@ -524,17 +526,44 @@ def _get_result(_source: Any, info: Info, **kwargs):
                 _source, info=info, args=field_args, kwargs=field_kwargs
             )
 
+        def wrap_field_extensions() -> Callable[..., Any]:
+            """Wrap the provided field resolver with the middleware."""
+
+            if field.extensions is None:
+                return _get_result
+
+            for extension in field.extensions:
+                extension.apply(field)
+
+            use_async = ensure_field_extension_compatibility(field)
```

Comment:
I might be mistaken here, but in case we have a field that supports both sync and async this would force the execution to call `resolve_async`, which would be wrong when using `schema.execute_sync`

Probably we need a flag here and wrap it 2 times:

- One for sync, which should also ensure that all extensions support `execute` (even the field itself)
- One for async that doesn't matter if it calls `resolve` or `resolve_async` because the compatibility itself will handle all incompatibilities.

Reply:
`use_async` only returns true if there is some extension or an async resolver that forces the use of async resolving. In case everything can be done sync (resolver is sync and extensions all support sync), `use_async` will always be `False`.